Mikado/utilities/file_type.py

```diff
@@ -1,18 +1,19 @@
 import sys
 
 magic_dict = {
-    "\x1f\x8b\x08": b"application/gzip",
-    "\x42\x5a\x68": b"application/x-bzip2",
-    "\x50\x4b\x03\x04": b"application/zip"
+    b"\x1f\x8b\x08": b"application/gzip",
+    b"\x42\x5a\x68": b"application/x-bzip2",
+    b"\x50\x4b\x03\x04": b"application/zip"
     }
 
+
 max_len = max(len(x) for x in magic_dict)
 
 
 def filetype(filename):
-    with open(filename) as f:
+    with open(filename, "rb") as f:
         file_start = f.read(max_len)
     for magic, ftype in magic_dict.items():
-        if file_start.startswith(magic):
+        if magic == file_start[:len(magic)]:
```

Comment:
Any reason why this is preferred  to file_start.startswith(magic), startswith conveys the intention in a much clearer way.

Reply:
Startswith does not function with bytes, so it's not an option :-)

Reply:
https://docs.python.org/3/library/stdtypes.html#bytes.startswith

Reply:
You are right, I must have typoed it when I tested! Amending and merging.